### ISSUE TITLE
[NavigationExperimental] Errors are swallowed on initial render

### ISSUE THREAD

**Issue by ronaldheft** [facebook/react-native#9075]
RN v0.30
iOS & Android

Using the `NavigationCardStack` component, any errors in the `renderScene` or `renderOverlay` will be swallowed on the initial render. For example:

```
<NavigationCardStack
  key="stack"
  navigationState={{ index: 0, routes: [{key: 'scene1'}] }}
  renderScene={props => {
    switch ( props.scene.route.key ) {
      case 'scene1':
        return <SomeComponent />;
      case 'scene2':
        return <SomeComponent2 />
    }
  }}
/>
```

Let's pretend `SomeComponent2` has some type of error that will cause it to fail to render.

On the initial render, all is well. Then we push to scene2 with `{key: 'scene2'}`. The red error screen appears. Great. This is all expected.

Now let's pretend `SomeComponent` has the error instead of `SomeComponent2`. In this scenario, on the initial render, the error is swallowed and we only see a blank screen. This is bad. No console errors appear either.

**Expected behavior:**

`NavigationCardStack` does not swallow error on the initial render.

**Comment by skatpgusskat** [comment]
i found there is missed type, ':' on second case (```case 'scene2':```). could it be matter?

**Comment by ronaldheft** [comment]
Above is pseudo code and not my specific example, so that was just a typo. I'll fix it when I get to a computer (can't edit on mobile).

**Comment by skatpgusskat** [comment]
@ronaldheft then could you share the full code? I want to debug with my environment.

**Comment by ronaldheft** [comment]
@skatpgusskat Yeah, I won't be able to do it today, but I'll put together a full sample of the issue this weekend.

**Comment by skatpgusskat** [comment]
@ronaldheft I create example with [this link](https://github.com/facebook/react-native/blob/c658cc545f0e8c508ab44389deed39801f4e472c/Libraries/CustomComponents/NavigationExperimental/NavigationCardStack.js) and make error with ```throw new Error()```, but even it was initial render, the red screen appear.

**Comment by ronaldheft** [comment]
I'm going to close this. I haven't exactly figured out the source of the issue yet, but I've been able to eliminate NavigationExperimental as the source.

Essentially, what's happening is my component render methods are somehow returning null on an error. I can try / catch the errors within each render method and then error reporting works fine, but if it's not caught within the component throwing the error, upstream components are not propagating the error.

I've isolated out NavigationExperimental by bypassing it in my project, and I saw the same problem of errors being swallowed, so NavigationExperimental is not the source. I'm still working to figure it out, and if I eliminate it down to something in React Native I'll open a new issue. Until then, assume this is something really strange with my environment.

Thanks for taking a look @skatpgusskat.
